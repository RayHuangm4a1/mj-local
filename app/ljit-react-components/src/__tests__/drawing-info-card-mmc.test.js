import React from 'react';
import { shallow, mount, } from 'enzyme';
import DrawingInfoCardMMC from '../components/drawing-info-card/elements/drawing-info-card-mmc';

jest.mock('../components/code-ball', () => {
	const MockComponent = () => <span />;

	MockComponent.ColorEnums = {
		GRADIENT_ORANGE: 'GRADIENT_ORANGE',
		LIGHTGREY: 'LIGHTGREY',
	};

	return {
		Round: MockComponent,
	};
});

describe('DrawingInfoCardMMC', () => {
	it('should handle default props', () => {
		const {
			className,
			buttonText,
			isButtonDisabled,
		} = DrawingInfoCardMMC.defaultProps;

		expect(className).toEqual('');
		expect(buttonText).toEqual('投注');
		expect(isButtonDisabled).toEqual(false);
	});

	it('should renders correctly', () => {
		const props = {
			opencode: '1,3,5,7,9',
			onClickButton: () => {},
		};
		const wrapper = shallow(<DrawingInfoCardMMC {...props} />);

		expect(wrapper).toMatchSnapshot();
	});

	it('should be selectable by class "drawing-info-card-mmc"', () => {
		const props = {
			opencode: '1,3,5,7,9',
			onClickButton: () => {},
		};
		const wrapper = shallow(<DrawingInfoCardMMC {...props} />);

		expect(wrapper.hasClass('drawing-info-card-mmc')).toEqual(true);
	});

	it('should disabled button when isButtonDisabled is true', () => {
		const props = {
			opencode: '1,3,5,7,9',
			onClickButton: () => {},
			isButtonDisabled: true,
		};
		const wrapper = shallow(<DrawingInfoCardMMC {...props} />);
		const button = wrapper.find('.drawing-info-card-mmc__bet');

		expect(button.props().disabled).toEqual(true);
	});

	it('should mount in a full DOM', () => {
		const props = {
			className: 'mock-class',
			opencode: '1,3,5,7,9',
			buttonText: 'mock-text',
			isButtonDisabled: false,
			onClickButton: () => {},
		};
		const wrapper = mount(<DrawingInfoCardMMC {...props} />);

		expect(wrapper.props().className).toEqual(props.className);
		expect(wrapper.props().opencode).toEqual(props.opencode);
		expect(wrapper.props().buttonText).toEqual(props.buttonText);
		expect(wrapper.props().isButtonDisabled).toEqual(props.isButtonDisabled);
		expect(wrapper.props().onClickButton).toEqual(props.onClickButton);
	});

	describe('when click button', () => {
		it('should onClickButton have been called', () => {
			const opencode = '1,3,5,7,9';
			const onClickButton = jest.fn();
			const wrapper = shallow(<DrawingInfoCardMMC opencode={opencode} onClickButton={onClickButton} />);
			const event = {
				preventDefault: () => {},
				target: { tagName: 'BUTTON', },
			};
			const button = wrapper.find('.drawing-info-card-mmc__bet');

			button.at(0).simulate('click', event);

			expect(onClickButton).toHaveBeenCalled();
			expect(onClickButton).toBeCalledWith(event);
		});

		it('should stop with disabled', () => {
			const opencode = '1,3,5,7,9';
			const isButtonDisabled = true;
			const onClickButton = jest.fn();
			const wrapper = mount(
				<DrawingInfoCardMMC
					opencode={opencode}
					onClickButton={onClickButton}
					isButtonDisabled={isButtonDisabled}
				/>
			);
			const button = wrapper.find('.drawing-info-card-mmc__bet');

			button.at(0).simulate('click');

			expect(onClickButton).toHaveBeenCalledTimes(0);
		});
	});
});
