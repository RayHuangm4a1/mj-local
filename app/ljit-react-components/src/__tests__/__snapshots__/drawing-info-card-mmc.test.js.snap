// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawingInfoCardMMC should renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DrawingInfoCardMMC
    buttonText="投注"
    className=""
    isButtonDisabled={false}
    onClickButton={[Function]}
    opencode="1,3,5,7,9"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <MockComponent
            className="drawing-info-card-mmc__ball"
            text="1"
          />
          <MockComponent
            className="drawing-info-card-mmc__ball"
            text="3"
          />
          <MockComponent
            className="drawing-info-card-mmc__ball"
            text="5"
          />
          <MockComponent
            className="drawing-info-card-mmc__ball"
            text="7"
          />
          <MockComponent
            className="drawing-info-card-mmc__ball"
            text="9"
          />
        </div>,
        <div
          className="drawing-info-card-mmc__bet-wrapper"
        >
          <button
            className="drawing-info-card-mmc__bet"
            disabled={false}
            onClick={[Function]}
          >
            投注
          </button>
        </div>,
      ],
      "className": "drawing-info-card-mmc",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <MockComponent
              className="drawing-info-card-mmc__ball"
              text="1"
            />,
            <MockComponent
              className="drawing-info-card-mmc__ball"
              text="3"
            />,
            <MockComponent
              className="drawing-info-card-mmc__ball"
              text="5"
            />,
            <MockComponent
              className="drawing-info-card-mmc__ball"
              text="7"
            />,
            <MockComponent
              className="drawing-info-card-mmc__ball"
              text="9"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1-0",
            "nodeType": "function",
            "props": Object {
              "className": "drawing-info-card-mmc__ball",
              "text": "1",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3-1",
            "nodeType": "function",
            "props": Object {
              "className": "drawing-info-card-mmc__ball",
              "text": "3",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "5-2",
            "nodeType": "function",
            "props": Object {
              "className": "drawing-info-card-mmc__ball",
              "text": "5",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "7-3",
            "nodeType": "function",
            "props": Object {
              "className": "drawing-info-card-mmc__ball",
              "text": "7",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "9-4",
            "nodeType": "function",
            "props": Object {
              "className": "drawing-info-card-mmc__ball",
              "text": "9",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="drawing-info-card-mmc__bet"
            disabled={false}
            onClick={[Function]}
          >
            投注
          </button>,
          "className": "drawing-info-card-mmc__bet-wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "投注",
            "className": "drawing-info-card-mmc__bet",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "投注",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <MockComponent
              className="drawing-info-card-mmc__ball"
              text="1"
            />
            <MockComponent
              className="drawing-info-card-mmc__ball"
              text="3"
            />
            <MockComponent
              className="drawing-info-card-mmc__ball"
              text="5"
            />
            <MockComponent
              className="drawing-info-card-mmc__ball"
              text="7"
            />
            <MockComponent
              className="drawing-info-card-mmc__ball"
              text="9"
            />
          </div>,
          <div
            className="drawing-info-card-mmc__bet-wrapper"
          >
            <button
              className="drawing-info-card-mmc__bet"
              disabled={false}
              onClick={[Function]}
            >
              投注
            </button>
          </div>,
        ],
        "className": "drawing-info-card-mmc",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <MockComponent
                className="drawing-info-card-mmc__ball"
                text="1"
              />,
              <MockComponent
                className="drawing-info-card-mmc__ball"
                text="3"
              />,
              <MockComponent
                className="drawing-info-card-mmc__ball"
                text="5"
              />,
              <MockComponent
                className="drawing-info-card-mmc__ball"
                text="7"
              />,
              <MockComponent
                className="drawing-info-card-mmc__ball"
                text="9"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1-0",
              "nodeType": "function",
              "props": Object {
                "className": "drawing-info-card-mmc__ball",
                "text": "1",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "3-1",
              "nodeType": "function",
              "props": Object {
                "className": "drawing-info-card-mmc__ball",
                "text": "3",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "5-2",
              "nodeType": "function",
              "props": Object {
                "className": "drawing-info-card-mmc__ball",
                "text": "5",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "7-3",
              "nodeType": "function",
              "props": Object {
                "className": "drawing-info-card-mmc__ball",
                "text": "7",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "9-4",
              "nodeType": "function",
              "props": Object {
                "className": "drawing-info-card-mmc__ball",
                "text": "9",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="drawing-info-card-mmc__bet"
              disabled={false}
              onClick={[Function]}
            >
              投注
            </button>,
            "className": "drawing-info-card-mmc__bet-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "投注",
              "className": "drawing-info-card-mmc__bet",
              "disabled": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "投注",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
